Flutter/calendar/appointments.md

```diff
@@ -130,6 +130,132 @@ events.notifyListeners(CalendarDataSourceAction.reset, null);
 {% endhighlight %}
 {% endtabs %}
 
+## DataSource manipulation
+
+By call this method whenever the datasource changes, to notify the datasource may have. Need to notify the datasource action(add, remove and reset) through notify listener.
+
+#### Add Action
+
+If the appointment is added in an underlying datasource, you can call the notifyListeners.
+
+In the following example, the appointment is added and notifyListeners is called in onPressed callback of the FloatingActionButton. 
+So, users can call notifyListeners methods to refresh the dataSource.
+
+{% tabs %}
+{% highlight dart hl_lines="14 15 16" %}
+
+@override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+          floatingActionButton: FloatingActionButton(
+            child: const Text('Add appointment'),
+            onPressed: () {
+              final Appointment app = Appointment(
+                  startTime: _controller.displayDate!,
+                  endTime:
+                      _controller.displayDate!.add(const Duration(hours: 2)),
+                  subject: 'Add Appointment',
+                  color: Colors.pink);
+              _events?.appointments!.add(app);
+              _events?.notifyListeners(
+                  CalendarDataSourceAction.add, <Appointment>[app]);
+            },
+          ),
+          body: SfCalendar(
+            view: CalendarView.month,
+            controller: _controller,
+            dataSource: _events,
+          )),
+    );
+  }
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Remove Action
+
+If the appointment is removed in an underlying datasource, you can call the notifyListeners.
+
+{% tabs %}
+{% highlight dart hl_lines="2 3 4" %}
+
+final Appointment removeAppointment = appointmentCollection[0];
+appointmentCollection.remove(removeAppointment);
+_events?.notifyListeners(CalendarDataSourceAction.remove,
+    <Appointment>[removeAppointment]);
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Reset Action
+
+If the appointment is replaced in an underlying datasource, you can call the notifyListeners.
+
+{% tabs %}
+{% highlight dart hl_lines="1 2 3" %}
+
+_events?.appointments!.clear();
+_events?.notifyListeners(
+    CalendarDataSourceAction.reset, _events!.appointments!);
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Add Resource
+
+If the resource is added in an underlying datasource, you can call the notifyListeners.
```

Comment:
Improve the content on remove, reset, addresource, remove resource and reset resource

Reply:
I have checked the content and updated the changes in latest commit.